File: backend/src/main/java/com/java/NBE4_5_1_7/global/dto/RsData.java
package com.java.NBE4_5_1_7.global.dto;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Getter;
import org.springframework.lang.NonNull;

@AllArgsConstructor
@Getter
@JsonInclude(JsonInclude.Include.NON_NULL)
public class RsData<T> {
    @NonNull
    private String code;
    @NonNull
    private String msg;
    @NonNull
    private T data;

    public RsData(String code, String msg) {
        this(code, msg, (T) new Empty());
    }

    @JsonIgnore
    public int getStatusCode() {
        String statusCodeStr = code.split("-")[0];
        return Integer.parseInt(statusCodeStr);
    }

}

File: backend/src/main/java/com/java/NBE4_5_3_7/global/dto/RsData.kt
package com.java.NBE4_5_3_7.global.dto;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;

@JsonInclude(JsonInclude.Include.NON_NULL)
data class RsData<T>(
    val code: String,
    val msg: String,
    val data: T
) {
    constructor(code: String, msg: String) : this(code, msg, Empty() as T)

    @get:JsonIgnore
    val statusCode: Int
        get() = code.split("-")[0].toInt()
}